File: mkdocs/overrides/devbook/accounts/create-from-mnemonic.mjs
import { Bip32 } from 'symbol-sdk';
import { SymbolFacade } from 'symbol-sdk/symbol';
import process from 'process';

// Initialize the facade for the testnet network [>step-1]
const facade = new SymbolFacade('testnet');
// [<step-1]
// Use an existing mnemonic if provided, otherwise generate a random one [>step-2]
const bip32 = new Bip32(facade.constructor.BIP32_CURVE_NAME);
let mnemonic = process.env.MNEMONIC;
if (mnemonic) {
	console.log('Loading mnemonic phrase from environment variable...');
} else {
	console.log('Generating random mnemonic phrase...');
	mnemonic = bip32.random();
}
console.log('Mnemonic phrase:', mnemonic);
// [<step-2]
// Load password from environment variable or use default [>step-3]
const password = process.env.PASSWORD || 'correcthorsebatterystaple';
console.log('Password:', password);

// Derive a root Bip32 node from the mnemonic and a password
const rootNode = bip32.fromMnemonic(mnemonic, password);
// [<step-3]
// Derive a child Bip32 node for the account at index 0 [>step-4]
const accountIndex = 0;
const childNode = rootNode.derivePath(facade.bip32Path(accountIndex));
// [<step-4]
// Convert the Bip32 node to a signing key pair [>step-5]
const keyPair = facade.constructor.bip32NodeToKeyPair(childNode);

// Derive the address from the public key
const address = facade.network.publicKeyToAddress(keyPair.publicKey);

// Output the account details
console.log('Address:', address.toString());
console.log('Public key:', keyPair.publicKey.toString());
console.log('Private key:', keyPair.privateKey.toString()); // [<step-5]

File: mkdocs/overrides/devbook/accounts/create-from-mnemonic.py
import os
from symbolchain.Bip32 import Bip32
from symbolchain.facade.SymbolFacade import SymbolFacade

# Initialize the facade for the testnet network
facade = SymbolFacade('testnet')

# Use an existing mnemonic if provided, otherwise generate a random one
bip32 = Bip32()
mnemonic = os.getenv('MNEMONIC')
if mnemonic:
	print("Loading mnemonic phrase from environment variable...")
else:
	print("Generating random mnemonic phrase...")
	mnemonic = bip32.random()
print(f'Mnemonic phrase: {mnemonic}')

# Load password from environment variable or use default
password = os.getenv('PASSWORD', 'correcthorsebatterystaple')
print(f'Password: {password}')

# Derive a root Bip32 node from the mnemonic and a password
root_node = bip32.from_mnemonic(mnemonic, password)

# Derive a child Bip32 node for the account at index 0
account_index = 0
child_node = root_node.derive_path(facade.bip32_path(account_index))

# Convert the Bip32 node to a signing key pair
key_pair = facade.bip32_node_to_key_pair(child_node)

# Derive the address from the public key
address = facade.network.public_key_to_address(key_pair.public_key)

# Output the account details
print(f'Address: {address}')
print(f'Public key: {key_pair.public_key}')
print(f'Private key: {key_pair.private_key}')
